festim/generic_simulation.py

```diff
@@ -136,8 +136,6 @@ def materials(self, value):
             self._materials = festim.Materials(value)
         elif isinstance(value, festim.Material):
             self._materials = festim.Materials([value])
-        elif isinstance(value, festim.Materials):
-            self._materials = value
```

Comment:
Hm I'd double check on this one.
Maybe it's better to have:
```python
        if isinstance(value, festim.Materials):
            self._materials = value
        elif isinstance(value, list):
            self._materials = festim.Materials(value)
        elif isinstance(value, festim.Material):
            self._materials = festim.Materials([value])

```

Reply:
I've double checked it :) Tests pass + manually verified. Alternatively, it'd be required to do what you suggested, yes. Maybe, your approach adds clarity for the code and there won't be any unexpected problems

Reply:
The one thing I'm worry about is what happens if we do:

```python
my_model.materials = F.Materials(....)
```
Theoretically, it is of type list (since it inherits from list). So, according to the code, it should perform:

```python
self._materials = festim.Materials(value)
```
but if value is already a festim.Materials class then any of the classe's attributes could be overwritten

I wonder what happens when you do:


```python

my_materials = F.Materials([1, 2, 3])
my_materials.D = 2 

my_model.materials = my_materials

print(my_model.materials.D)  # does it print 2 or does it print the default value of the argument None?
```

Reply:
Well, it's a very good point. It returns `None`. Your approach returns `2`. Btw, I've included it in the latest commit

Reply:
Ha! Knew it! Maybe we want to write a test for this then since it wasn't caught by the CI.

Something like:

```python


@pytest.mark.parametrize("input,expected_output", [(test_materials, test_materials) ])
def test_materials_setter():
   my_model = F.Model()
   test_materials = F.Materials([])
   my_model.materials = test_materials 
   assert my_model.materials == test_materials
```

I know it sounds silly but still good to test!

Reply:
@KulaginVladimir I edited my code I initially put a parametrize mark but no need

Reply:
Modified this